odoo/addons/base/ir/ir_http.py

```diff
@@ -320,6 +320,11 @@ def binary_content(cls, xmlid=None, model='ir.attachment', id=None, field='datas
             if not mimetype:
                 mimetype = guess_mimetype(base64.b64decode(content), default=default_mimetype)
 
+        # extension
+        extension = mimetypes.guess_extension(mimetype)
+        if not filename.endswith(extension):
```

Comment:
According to its documentation, `guess_extension` can return nothing (if the system doesn't have a mapping for the provided mimetype). Also it looks like the filename could still be empty for e.g. an object with a name / filename_field which was left empty. Maybe the filename section should fallback the same way mimetype does? (e.g. `if not filename and …` instead of `elif/else`).

Reply:
@xmo-odoo Changes are done. Can you please check it now?